docs/install-upgrade.md

```diff
@@ -155,12 +155,47 @@ To install the MongoDB Community Kubernetes Operator:
 
 ## Upgrade the Operator
 
-To upgrade the MongoDB Community Kubernetes Operator to a specific version:
+The release v0.6.0 had some breaking changes (see https://github.com/mongodb/mongodb-kubernetes-operator/releases/tag/v0.6.0) requiring special steps to upgrade from a pre-0.6.0 Version.
+As always, have backups.
+Make sure you run commands in the correct namespace.
 
-1. Change the version in the Operator [resource definition](../config/manager/manager.yaml)
-2. Change to the directory in which you cloned the repository.
-3. Checkout the specific tag matching the operator version (e.g. `v0.5.1`)
-4. Invoke the following `kubectl` command to upgrade the [Custom Resource Definitions](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/).
-   ```
-   kubectl apply -f config/crd/bases/mongodbcommunity.mongodb.com_mongodbcommunity.yaml
-   ```
+1. Prepare for the upgrade.
+
+   a. Migrate your cr by updating apiVersion and kind to
+      ```
+      apiVersion: mongodbcommunity.mongodb.com/v1
+      kind: MongoDBCommunity
+      ```
+      If you upgrade from pre-0.3.0 you need to also add the field spec.users[n].scramCredentialsSecretName for each resource. This will be used to determine the name of the generated secret which stores MongoDB user credentials. This field must comply with DNS-1123 rules (must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character).
+   b. Plan a downtime.
+2. Out with the old
+
+   a. Delete the old operator.
+      ```
+      kubectl delete deployments.apps mongodb-kubernetes-operator
+      ```
+   b. Delete the old statefulset.
+      ```
+      kubectl delete statefulsets.apps mongodb
+      ```
+   c. Delete the old customResourceDefinition. Not strictly needed but no need to keep it around anymore (unless you got more installations of operator in your cluster)
+      ```
+      kubectl delete crd mongodb.mongodb.com
+      ```
+3. In with the new
+
+   Follow the normal installation procedure above.
+4. Start up your cluster again
+   a. Re-create your cr using the new Version from Step 1.a
+   b. Patch your statefulset to have it update the permissions
```

Comment:
Is this something specific you need for your environment? The Operator should create the StatefulSet with the right SecurityContext at this stage, and I'm not sure why you would need to patch it manually.

Reply:
The patch is from release notes of v0.6.0 to fix filesystem rights on the PV.
It is actually needed for mongod to be able to start with the old data.